src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -280,7 +280,12 @@ private Socket TryConnectParallel(string hostName, int port, TimeSpan ts, bool i
             Task<Socket> connectTask;
 
             Task<IPAddress[]> serverAddrTask = Dns.GetHostAddressesAsync(hostName);
```

Comment:
When I add ConfigureAwait(false), I get a ConfiguredTaskAwaitable. I don't see a way to wait for that to complete within a specified duration.

When I was looking at the original code, it looked like the original writer was trying to guard against DNS taking too long and going beyond our timeout. It calls GetHostAddressesAsync, waits for the task with Wait(ts), but then calls serverAddrTask.Result. Doesn't calling .Result on the task just block until completion anyway? It seems like it would have been more efficient to just call GetHostAddressesAsync(hostName).GetAwaiter().GetResult() instead of blocking a thread with the Wait(ts) call.

Anyways, if we aren't worried about DNS taking too long, (DNS timeouts are generally ~2s) it might be more efficient to simply change them to GetHostAddressesAsync(hostName).GetAwaiter().GetResult() and get rid of the attempt at not letting DNS exceed our timeout.

Reply:
Using ConfigureAwait(false) would be effective on `asyn-await` approach. If there is any plan to improve the code using this approach, I would suggest keeping it as is.